resources/qml/dialogs/UserProfile.qml

```diff
@@ -292,13 +300,13 @@ ApplicationWindow {
             Image {
                 Layout.preferredHeight: 16
                 Layout.preferredWidth: 16
-                source: ((model.verificationStatus == VerificationStatus.VERIFIED) ? "image://colorimage/:/icons/icons/ui/lock.png?green" : ((model.verificationStatus == VerificationStatus.UNVERIFIED) ? "image://colorimage/:/icons/icons/ui/unlock.png?yellow" : "image://colorimage/:/icons/icons/ui/unlock.png?red"))
+                source: ((model.verificationStatus == VerificationStatus.VERIFIED) ? "image://colorimage/:/icons/icons/ui/lock.png?green" : ((model.verificationStatus == VerificationStatus.UNVERIFIED) ? "image://colorimage/:/icons/icons/ui/unlock.png?yellow" : ((model.verificationStatus == VerificationStatus.SELF) ? "image://colorimage/:/icons/icons/ui/checkmark.png?green" : "image://colorimage/:/icons/icons/ui/unlock.png?red")))
             }
 
             Button {
                 id: verifyButton
 
-                visible: (!profile.userVerificationEnabled && !profile.isSelf) || (profile.isSelf && (model.verificationStatus != VerificationStatus.VERIFIED || !profile.userVerificationEnabled))
+                visible: (!profile.userVerificationEnabled && !profile.isSelf) || (profile.isSelf && (model.verificationStatus == VerificationStatus.UNVERIFIED || !profile.userVerificationEnabled))
```

Comment:
I used the != VERIFIED originally, because there is a TOFU state, but I guess for yourself that does not matter.